test/URLEndpointListenerTest.cc

```diff
@@ -205,4 +205,89 @@ TEST_CASE_METHOD(URLEndpointListenerTest, "Listener with OneShot Replication", "
     CBLURLEndpointListener_Free(listener);
 }
 
+TEST_CASE_METHOD(URLEndpointListenerTest, "Listener with Basic Authentication", "[URLListener]") {
+    struct Context {
+        int rand = 6801;
+    } context;
+
+    static constexpr slice kUser{"pupshaw"};
+    static constexpr slice kPassword{"frank"};
+
+    CBLURLEndpointListenerConfiguration listenerConfig {
+        &context, // context
+        cy.data(),
+        2,
+        0
+    };
+    listenerConfig.disableTLS = true;
+
+    CBLListenerAuthenticator* auth = nullptr;
+    SECTION("Successful Login") {
+        auth = CBLListenerAuth_CreatePassword([](void* ctx, FLString usr, FLString psw) {
+            auto context = reinterpret_cast<Context*>(ctx);
+            CHECK(context-> rand  == 6801);
+            return usr == kUser && psw == kPassword;
+        });
+        listenerConfig.authenticator = auth;
+        auth = nullptr; // ownership passed to listenerConfig
```

Comment:
The ownership is not passed to the listener config. But the CBLURLEndpointListener should copy the auth instead. At least, this is the way we have with the Replicator's authenticator.

Reply:
We can discuss more on this. Alternatively, we can make it RefCounted as well.